modules/renter/download.go

```diff
@@ -549,8 +611,7 @@ func (r *Renter) managedWaitOnDownloadWork(ds *downloadState) {
 	// If the chunk has completed, perform chunk recovery.
 	if len(cd.completedPieces) == cd.download.erasureCode.MinPieces() {
 		err := cd.recoverChunk()
-		ds.activePieces -= len(cd.completedPieces)
-		cd.completedPieces = make(map[uint64][]byte)
```

Comment:
Thanks for letting me know. I was thinking about rewriting this file but don't have the time right now. Maybe we could discuss possible changes to this at some point in the coming weeks. In the meantime, I'll add a couple of tests to check these cases.